### ISSUE TITLE
Tary problem:right click Tray ico eject the error''TypeError:menu.groupsMap[id].find is not a function"

### ISSUE THREAD

**Issue by Iteravse** [electron/electron#23071]
### Issue Details

* **Electron Version:**
  *  5.0.8 
* **Operating System:**
  *  Windows 7 (Sp1)  x64 -->

### Expected Behavior
i just want to build a Tray on task bar,i throughly according the 
[official Document ](https://www.electronjs.org/docs/api/tray)to append the codes in main.js,and it worked,but having litte flaw
### Actual Behavior
it actually question is when i right click the Tray ico,it eject the error certainlly like these

```
A JavaScript error occurred in the main process
Uncaught Exception
   TypeError:menu.groupsMap[id].find is not a function
    at Function .menuWillShow
    myProjectLocate\node_modules\electron\dist\...:40)
```
### To Reproduce
There is my case,i wirte these code in Event when i hide the window,to realize the function about Minimize App to Tray:

```
 mainWindow.hide();
  appTray = new Tray(path.join(AppLoc, '/img/MindMaster.ico'));
  const contextMenu = Menu.buildFromTemplate([
    { label: 'Item1', type: 'radio' },
    { label: 'Item2', type: 'radio' },
    { label: 'Item3', type: 'radio', checked: true },
    { label: 'Item4', type: 'radio' }
  ])
  appTray.setToolTip('This is my application.');
  appTray.setContextMenu(contextMenu);
```
  it worked, when i click the button window will hide,and the Tary ico will appeared,but when i right click that ico,it must eject the error like preceding description:

> "menu.groupsMap[id].find is not a function"

  i try to annotate the code'mainWindow.hide();',and set the mainwindow don't fullscreen,but the problem still can't solve,but the menu context will eject after confirm the error,any idear or suggest?,it seems not my coding issue:)

**Comment by Iteravse** [comment]
it seems electron runtime evironment problem,becasue,when i update the electron version with 8.1.2,this question solved,(5.0.8 and 6.1.9 try these version it is can't solve)but i encounter aother question,some animation effect are very lags in new version with same code cases , what's worng with new version of electron?,what's changing about animation for new version?,by the way i using ‘requestAnimationFrame’ to Rending the animation,it's smooth on 5.0.8 but very lag in 8.12,have any setting can solve the phenomenon?if not,so how to solve the Tray icon right click poping error?

**Comment by electron-triage** [comment]
The Electron version reported on this issue is no longer supported. See our [supported versions documentation](https://www.electronjs.org/docs/tutorial/support#supported-versions).

**If you're still experiencing this issue on a supported version, please open a new issue with an updated repro - a [Fiddle](https://www.electronjs.org/fiddle) is very appreciated.**

Electron has a large issues backlog. To help our team prioritize, we're closing older issues and asking for new issues with updated repro steps if it affects a supported version. This helps sort what issues are still relevant and helps us fix them more quickly.

Thanks for your patience and understanding!
